### PR TITLE
pre-commit config & black/Flake8/mypy fixes

### PR DESCRIPTION
**Pull request by jpopelka**

### CODE REVIEW

File: ogr/services/github.py [hunk 1]
```diff
@@ -209,6 +209,8 @@ def get_releases(self) -> List[Release]:
 
 
 class GithubUser(BaseGitUser):
+    service: GithubService
+
```
Comment (jpopelka) [inline ogr/services/github.py]: previously the class looked like:
```python
class GithubUser(BaseGitUser):
    def __init__(self, service: GithubService) -> None:
        super().__init__(service=service)

    def get_username(self) -> str:
        return self.service.github.get_user().login
```
and mypy kept telling me that:
`error: "GitService" has no attribute "github"`
while pointing at the
`return self.service.github.get_user().login`
line.

I resolved it by declaring this instance variable.
@lachmanfrantisek do you have any better idea?

File: ogr/abstract.py [hunk 2]
```diff
@@ -320,7 +320,7 @@ def get_file_content(self, path: str, ref="master") -> str:
 
 
 class GitUser:
-    def __init__(self, service: "GitService") -> None:
+    def __init__(self, service: GitService) -> None:
```
Comment (lachmanfrantisek) [inline ogr/abstract.py]: I think this can cause import problems since GitService is in the same file.

File: ogr/services/github.py (hunk 1 above)
Comment (lachmanfrantisek) [reply ogr/services/github.py]: Isn't this a class attribute or is this used only for the typing?

Comment (jpopelka) [reply ogr/services/github.py]: It's just hint/annotation, see https://www.python.org/dev/peps/pep-0526/#class-and-instance-variable-annotations

File: ogr/abstract.py (hunk 2 above)
Comment (jpopelka) [reply ogr/abstract.py]: I thought we need to use the quotations only
* in the same class
* in the same module if the type/class is not yet defined

In this case, the `GitService` is defined on line 78, while this is line 323.

I can change it back, but I'd like to understand it.

Comment (lachmanfrantisek) [reply ogr/abstract.py]: Yes, you're right. It's already defined so it's not needed.

**Pull request merged**